driver_test.go

```diff
@@ -0,0 +1,104 @@
+package sqlair_test
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"strings"
+	"sync"
+	"unsafe"
+
+	"github.com/mattn/go-sqlite3"
+)
+
+// This file contains a wrapper sql.Driver over the SQLite driver which
+// monitors the creation and closing of prepared statements and stores the
+// references to said statements. We can later use that information to check
+// for statement leaks.
+
+// The stmt registry keeps the pointers for the open and closed statements to
+// detect resource leaks. It uses unsafe pointers instead of references to the
+// object because if we stored a reference the runtime.Finalizer would not be
+// able to run.
```

Comment:
This comment should explain why `openedStmts` and `closedStmts` are maps of maps (i.e. because they are per test).